tests/utils.go

```diff
@@ -158,6 +167,14 @@ func createDistributionManager() {
 	testsBundleDeleteRemoteService.DistDetails = distDetails
 }
 
+func createXrayManager() {
+	xrayDetails := GetXrayDetails()
+	client, err := rthttpclient.ArtifactoryClientBuilder().SetServiceDetails(&xrayDetails).Build()
```

Comment:
Hey @eyalbe4,

I understand why you want to move that functionality, but I've noticed a few things.

- In the root directory, there already is a generic [httpclient](https://github.com/jfrog/jfrog-client-go/tree/master/httpclient) package, which [ArtifactoryHttpClient](https://github.com/jfrog/jfrog-client-go/blob/master/artifactory/httpclient/httpclient.go#L15) depends on.
- The Distribution manager is also depend on the [ArtifactoryClientBuilder](https://github.com/jfrog/jfrog-client-go/blob/master/distribution/manager.go#L4)

This sounds like a big change that should go in a separate PR, as it would effect many files under `artifactory/*`, `distribution/*`, `tests/*` and `xray/*`.

As the `httpclient` directory & package is already taken, what did you want to do?

Cheers

Reply:
Got it @josh-barker-coles. We'll create an issue for this and handle this in a follow up PR.